pkg/apis/componentconfig/types.go

```diff
@@ -474,6 +474,11 @@ type KubeControllerManagerConfiguration struct {
 	HorizontalPodAutoscalerSyncPeriod unversioned.Duration `json:"horizontalPodAutoscalerSyncPeriod"`
 	// deploymentControllerSyncPeriod is the period for syncing the deployments.
 	DeploymentControllerSyncPeriod unversioned.Duration `json:"deploymentControllerSyncPeriod"`
+	// apiServerHealthCheckPeriod is the period for checking the health of the API servers.
+	// If <= 0, the API server health checks are disabled.  In order to work, the
+	// controller manager must be able to reach all of the API servers as specified in the
+	// "kubernetes" auto-created endpoints list.
+	APIServerHealthCheckPeriod unversioned.Duration `json:"apiServerHealthCheckPeriod"`
```

Comment:
"Health check" is ambiguous, which is why we don't use it in the API.

This sounds like a readiness probe. I suggest calling it that instead.